### PR TITLE
[CELEBORN-1052][FOLLOWUP] Introduce dynamic ConfigService at SystemLevel and TenantLevel

### PR DESCRIPTION
**Pull request by SteNicholas**
### What changes were proposed in this pull request?

Follow up #2100. Mainly changes the package from scala to java of the codes in #2100. Meanwhile, `FsConfigServiceImpl#refresh` should directly return instead of refreshing configs. 

### Why are the changes needed?

This PR follow up dynamic `ConfigService` at `SystemLevel` and `TenantLevel`, Dynamic configuration is a type of configuration that can be changed at runtime as needed in #2100. The implementation of `ConfigService` is based on Java codes, which are put into Scala package and cause that the spotless plugin does not format well. After the changes of the pull request, there are much code style changes generated from the package moving behavior.

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

`ConfigServiceSuiteJ`.

### CODE REVIEW

**Comment by pan3793** [comment]
why do moving files produce lots of code style changes?

**Comment by SteNicholas** [comment]
@pan3793, the implementation of `ConfigService` is based on Java codes, which are put into Scala package and the format does not work.

**Comment by pan3793** [comment]
> @pan3793, the implementation of `ConfigService` is based on Java codes, which are put into Scala package and the format does not work.

alright, put such information to "Why are the changes needed?"

File: service/src/main/java/org/apache/celeborn/server/common/service/config/ConfigService.java [hunk 1]
```diff
@@ -17,7 +17,7 @@
 
 package org.apache.celeborn.server.common.service.config;
 
-public interface ConfigService {
+public interface ConfigService extends AutoCloseable {
```
Comment (pan3793) [inline service/src/main/java/org/apache/celeborn/server/common/service/config/ConfigService.java:20]: looks unnecessary change

Comment (pan3793) [reply service/src/main/java/org/apache/celeborn/server/common/service/config/ConfigService.java:20]: alright, no strong opinion here

File: service/src/main/java/org/apache/celeborn/server/common/service/config/FsConfigServiceImpl.java [hunk 2]
```diff
@@ -88,10 +85,12 @@ private synchronized void refresh() {
       }
     } catch (Exception e) {
       LOG.warn("Refresh dynamic config error: {}", e.getMessage(), e);
+      return;
     }
 
     tenantConfigAtomicReference.set(tenantConfs);
-    systemConfigAtomicReference.set(systemConfig == null ? new SystemConfig(celebornConf) : systemConfig);
+    systemConfigAtomicReference.set(
+        systemConfig == null ? new SystemConfig(celebornConf) : systemConfig);
```
Comment (pan3793) [inline service/src/main/java/org/apache/celeborn/server/common/service/config/FsConfigServiceImpl.java:93]: why not initialize the systemConfig on defining, then we can also `null` propagation

File: service/src/main/java/org/apache/celeborn/server/common/service/config/FsConfigServiceImpl.java [hunk 3]
```diff
@@ -88,10 +85,12 @@ private synchronized void refresh() {
       }
     } catch (Exception e) {
       LOG.warn("Refresh dynamic config error: {}", e.getMessage(), e);
+      return;
```
Comment (pan3793) [inline service/src/main/java/org/apache/celeborn/server/common/service/config/FsConfigServiceImpl.java:88]: this silently changes the logic, but looks fine

File: service/src/main/java/org/apache/celeborn/server/common/service/config/FsConfigServiceImpl.java (hunk 2 above)
Comment (RexXiong) [reply service/src/main/java/org/apache/celeborn/server/common/service/config/FsConfigServiceImpl.java:93]: Actually, the current design treats CelebornConf as the baseConf for SystemConfig, so that the interfaces of ConfigServer can be unified to DynamicConf.  Of course, it's also possible to have SystemConfig as null, but that would require ConfigService to implement other interfaces such as getSystemValue/getTenantValue. However, this may not easy to make a configuration cache. Therefore, I adopted the first design.

File: service/src/main/java/org/apache/celeborn/server/common/service/config/FsConfigServiceImpl.java (hunk 3 above)
Comment (SteNicholas) [reply service/src/main/java/org/apache/celeborn/server/common/service/config/FsConfigServiceImpl.java:88]: @RexXiong, @pan3793, I have updated the `What changes were proposed in this pull request?` to mention this change.

**Review by pan3793 — APPROVED** [review]

File: service/src/main/java/org/apache/celeborn/server/common/service/config/FsConfigServiceImpl.java (hunk 3 above)
Comment (RexXiong) [reply service/src/main/java/org/apache/celeborn/server/common/service/config/FsConfigServiceImpl.java:88]: I rethought that this change might cause systemConf to be null

**Review by RexXiong — APPROVED** [review]
LGTM, merge to main(v0.4.0)
